hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/client/HddsClientUtils.java

```diff
@@ -127,17 +127,25 @@ private static void doNameChecks(String resName) {
     }
   }
 
-  private static boolean isSupportedCharacter(char c) {
-    return (c == '.' || c == '-' ||
-        Character.isLowerCase(c) || Character.isDigit(c));
+  private static boolean isSupportedCharacter(char c, boolean isNamespaceS3RuleCompliant) {
+    if (isNamespaceS3RuleCompliant) {
+      return (c == '.' || c == '-' ||
+              Character.isLowerCase(c) || Character.isDigit(c));
+    } else {
+      // allow ozone namespace to follow other volume/bucket naming convention,
+      // for example, here adds 'underscore',
+      // which is a valid character in POSIX-compliant system, like HDFS.
+      return (c == '.' || c == '-' || c == '_' ||
+              Character.isLowerCase(c) || Character.isDigit(c));
+    }
```

Comment:
This can be simplified:

```suggestion
    return (c == '.' || c == '-' ||
        Character.isLowerCase(c) || Character.isDigit(c)) ||
        (c == '_' && !isNamespaceS3RuleCompliant);
    }
```

Reply:
Yup updated thanks attila!